.clang-tidy

```diff
@@ -93,14 +101,15 @@ WarningsAsErrors: '*'
 #   - misc-no-recursion                                         Ignore general recommendation to avoid recursion, which we commonly use when working with query plans.
 #   - bugprone-easily-swappable-parameters                      Ignore issues with swappable parameters as we found them to be of no help.
 #   - bugprone-suspicious-include                               Unity builds use `#include`s for cpp files. However, unity builds significantly improve the runtime of clang-tidy,
-#                                                                which is otherwise the slowest step of the CI pipeline by a factor of two. Furthermore, the linter checks for 
+#                                                                which is otherwise the slowest step of the CI pipeline by a factor of two. Furthermore, the linter checks for
 #                                                                these includes as well.
 #   - bugprone-unchecked-optional-access                        Too many false positives (with clang-tidy 14/15). TODO(anyone): Re-evaluate for future clang versions.
 #   - llvm-header-guard                                         We prefer the simpler `#pragma once`. The main arguments (e.g., lacking support in old compilers)
 #                                                               do not matter with our limited set of supported compilers.
 #   - boost-use-ranges                                          As most boost-ranges implementations are now in C++20, we prefer `std::ranges`.
 #   - portability-template-virtual-member-function              We do not mind being incompatible with the MSVC compiler.
-
+#   - cppcoreguidelines-avoid-const-or-ref-data-members         We have quite a few const and ref data members. This is a bad thing, but it would take quite a while to fix it all.
```

Comment:
I don't agree that those are bad. The reason tidy does not want them is that we might not recognize if we screw up move constructors. Totally valid point, but as long as we ensure this will not happen, we should be fine, right?

Reply:
Maybe something to discuss in person. I like const members just as much as I like this tidy rule.

Reply:
Yes.